src/visitors.rs

```diff
@@ -83,6 +83,20 @@ impl<'a, 'b: 'a, 'tcx: 'b> MirVisitor<'a, 'b, 'tcx> {
         }
     }
 
+    /// Restores the method only state to its initial state.
+    fn reset_visitor_state(&mut self) {
```

Comment:
It kind of sucks that you have to repeat this. It would have been nice if all of these were under a substruct, and both here and in the new() you could just set it to ::default. Not sure how you feel about that.

Reply:
I am of mind to do that, but only when I have some spare time and an empty PR queue.